<?xml version="1.0"?>
<launch>
  <!-- start the teleop app -->
  <arg name="namespace" default="/merbots/wrov/rov"/>
  <arg name="log2Console" default="true"/>
  <arg name="port" default="service"/>
  <!--to use real set port to "/dev/ttyUSB0" -->
  <arg name="dccommsId" default="g500_s100"/>
  <!-- used when port == "service" -->
  <arg name="log2File" default="false"/>
  <arg name="launchComms" default="true"/>
  <arg name="launchImage" default="true"/>
  <arg name="image" default="/uwsim/camera1"/>
  <arg name="imageTransport" default="raw"/>
  <arg name="encodedImage" default="encoded_image"/>
  <arg name="simrobot" default="true"/>
  <arg name="enableSafety" default="false" />
  <arg name="enableRviz" default="false" />
  <arg name="enableShadders" default="true" />
  <arg name="sceneXml" default="netsim_cirs.xml" />
  <arg name="enableUWSim" default="true" />

  <group ns="$(arg namespace)">
    <node if="$(arg launchComms)" pkg="wireless_ardusub" type="rov_v3" name="rov_v3" output="screen" required="true">
      <param name="port" value="$(arg port)" type="string"/>
      <param name="log2Console" value="$(arg log2Console)" type="bool"/>
      <param name="log2File" value="$(arg log2File)" type="bool"/>
      <param name="dccommsId" value="$(arg dccommsId)" type="string"/>
    </node>
  </group>
  <group if="$(arg launchImage)">
    <include file="$(find image_utils_ros)/launch/encoder.launch">
      <arg name="image" value="$(arg image)"/>
      <arg name="encodedImage" value="$(arg encodedImage)"/>
      <arg name="imageTransport" value="$(arg imageTransport)"/>
      <arg name="namespace" value="$(arg namespace)"/>
    </include>
  </group>

  <group if="$(arg simrobot)">
	<include file="$(find cola2_g500)/launch/sim_g500_start.launch">
		<arg name="enableRviz" value="false"/>
		<arg name="enableSafety" value="$(arg enableSafety)"/>
		<arg name="enableJoystick" value="false"/>
	</include>
  </group>



</launch>
